frontend/dispatcher.lua

```diff
@@ -618,7 +618,7 @@ function Dispatcher:execute(ui, settings, gesture)
     for k, v in pairs(settings) do
         if settingsList[k] ~= nil and (settingsList[k].conditions == nil or settingsList[k].conditions == true) then
             -- Be sure we don't send a document setting event if there's not yet or no longer a document
-            if (not settingsList[k].paging and not settingsList[k].rolling) or self.document then
+            if self.document or (not settingsList[k].paging and not settingsList[k].rolling) then
```

Comment:
You're sure it works with `self.document` ?  The other reference in this file is `ui.document`.

Reply:
Yes I have tripple checked: With `self.document` and `ui.document` the patch works. With none of them I get the crash.

So for consistancy I think we should use `ui.document` here too.

Reply:
Well, I don't see anywhere some code that would set or update `Dispatcher.document` :/
So, best to go with `ui.document` (as `ui` is provided to the function call, so we're sure it's up to date.